### PR TITLE
feat(app); [WEB-4] created Navbar conponent & Dropdown conponent

### PR DESCRIPTION
**Pull request by EduardKurbatov**
Was created navbar component and dropdown conponent.bouth components are adaptive for every screen resolution

### CODE REVIEW

File: src/components/NavBar/NavBar.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import "./NavBar.scss";
+import { useState } from "react";
+import { Link } from "react-router-dom";
+import Dropdown from "./Dropdown";
+
+export const NavBar = () => {
+  const [click, setClick] = useState(false);
+  const [dropdown, setDropdown] = useState(false);
+
+  const handleClick = () => setClick(!click);
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.js:10]: It's better to rename `handleClick` to `toggleMobileMenu` for more code readability

File: src/components/NavBar/NavBar.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import "./NavBar.scss";
+import { useState } from "react";
+import { Link } from "react-router-dom";
+import Dropdown from "./Dropdown";
+
+export const NavBar = () => {
+  const [click, setClick] = useState(false);
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.js:7]: Please rename `click`and `setClick` to `isMenuActive` and `setMenuState` respectively

File: src/components/NavBar/NavBar.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  const [dropdown, setDropdown] = useState(false);
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.js:8]: This one should be renamed to `isDropdownActive` and `setDropdownVisibility` respectively

File: src/components/NavBar/NavBar.scss [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    justify-content: end;
+    margin-right: 2rem;
+    .nav-item {
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+      margin-left: 15px;
+      font-size: 12px;
+      .nav-links {
+        color: #999;
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.scss:43]: For not hardcoding and copy-pasting this color you can create a variable in `variables.scss` and just import it in every needed location

File: src/components/NavBar/recovery [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                <li><a href="#">Service3</a></li>
+              </ul>
+            </li>
+          </ul>
+        </li>
+      </ul>
+  );
+}
+
+export default Dropdown;
```
Comment (VeeShort) [inline src/components/NavBar/recovery:28]: Holy shit dude, what is that? Remove it before the Universe collapses

**Review by VeeShort — CHANGES_REQUESTED** [review]
Few comments, also change every double-quote to single-quote in imports

**Comment by VeeShort** [comment]
PR title should be `feat(app): Navbar and Dropdown components`

**Review by VeeShort — CHANGES_REQUESTED** [review]
PR title should be `feat(app): [WEB-4] Navbar and Dropdown components`
Check spelling in the description

File: src/components/NavBar/Dropdown.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import "./Dropdown.scss";
+
+function Dropdown() {
+  const [closeSub, setSubState] = useState(false);
+  const [closeSubSub, setSubSubState] = useState(false);
+
+  const changeSubState = e => {
+    e.preventDefault();
+    setSubState(!closeSub);
+    console.log(closeSub);
```
Comment (VeeShort) [inline src/components/NavBar/Dropdown.js:11]: Remove console.log

Comment (VeeShort) [inline src/components/NavBar/Dropdown.js:11]: remove console.log

File: src/components/NavBar/Dropdown.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  };
+
+  const changeSubSubState = e => {
+    e.preventDefault();
+    setSubSubState(!closeSubSub);
+    console.log(closeSubSub);
```
Comment (VeeShort) [inline src/components/NavBar/Dropdown.js:17]: remove console.log

**Review by VeeShort — CHANGES_REQUESTED** [review]

**Comment by VeeShort** [comment]
@EduardKurbatov before you fix merge conflicts, run `npm i`

File: src/components/NavBar/Dropdown.scss [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  .subMenu_2 {
+    visibility: visible;
+  }
+}
+
+.active-link_2:hover {
+  .subMenu_3 {
+    visibility: visible;
+  }
+}
```
[the remark this answers is not in the document]
Comment (EduardKurbatov) [reply src/components/NavBar/Dropdown.scss:63]: i need other solution here.Cuz sub-menu items are child elements for the first sub-menu item.If I try to do like on your example all child sub-menu items will be showed at the same time

File: src/components/NavBar/NavBar.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    setDropdown(!dropdown)
+  }
+
+  const onMouseLeave = () => {
+    if (window.innerWidth < 960) {
+      setDropdown(false);
+    } else {
+      setDropdown(false);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (EduardKurbatov) [reply src/components/NavBar/NavBar.js:31]: It's not true.Cuz if i clear the "else" part first sum-menu item will never hide

File: src/components/NavBar/recovery (hunk 5 above)
Comment (EduardKurbatov) [reply src/components/NavBar/recovery:28]: lol.I'm so sorry about that)

File: src/components/NavBar/Dropdown.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (VeeShort) [reply src/components/NavBar/Dropdown.scss:63]: @EduardKurbatov I see, then we can try this solution:
```css
#active-link_1:hover,
#active-link_2:hover {
  & > .sub-menu { visibility: visible; }
}
```

**Review by VeeShort — CHANGES_REQUESTED** [review]

File: src/components/NavBar/Dropdown.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+  const changeSubState = e => {
+    e.preventDefault();
+    setSubState(!closeSub);
+  };
+
+  const changeSubSubState = e => {
+    e.preventDefault();
+    setSubSubState(!closeSubSub);
+  };
```
Comment (VeeShort) [inline src/components/NavBar/Dropdown.js:16]: I think we can change this logic to this:
```js
const [list, setList] = useState([false, false, false]);

const  clickHandler = (index) => {
  list[index] = !list[index];
  setList(list);
}
```
and then change `onClick`  in `render()`: 
```jsx
// ...
<span onClick={() => clickHandler(0)}>
// ...
<span onClick={() => clickHandler(1)}>
// and so on
```
and className check for the same `<span>` (this change will require to change hide/show logic in css):
```jsx
className={list[0] && 'active'}
```

File: src/components/NavBar/NavBar.scss [hunk 11]
```diff
@@ -0,0 +1,126 @@
+$font-color-mobile: #fff;
+$font-color: #999;
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.scss:2]: The same variable is already used in `Dropdown.scss`, create a separate file called `variables.scss` and put every variable there, then you can just  __@include__ it in places you need

File: src/components/NavBar/NavBar.scss [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+      height: auto;
+    }
+
+    .navbar-logo {
+      color: $font-color-mobile;
+    }
+    .nav-menu {
+      visibility: hidden;
+      position: absolute;
+      display: flex;
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.scss:83]: `.nav-menu` already has this property defined

File: src/components/NavBar/NavBar.scss [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+      .ion-chevron-up {
+        display: none;
+      }
+      .nav-links {
+        border-bottom: 1px solid rgba(255, 255, 255, 0.05);
+        text-align: left;
+        color: $font-color-mobile;
+        width: 100%;
+      }
+    }
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.scss:114]: Move this css to `.nav-menu` and mention it like:
```scss
&.active {
  ...
}
```
Also remove next properties for `.active`:
`left, top` as they are already defined in `.nav-menu` and does not change for `.active`

File: src/components/NavBar/NavBar.scss [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+
+@media screen and (max-width: 960px) {
+  .container {
+    width: 100%;
+    height: 70px;
+    background: #0289ff;
+    z-index: 25;
+    padding: 0;
+
+    .navbar .nav-menu .nav-item.mobileNav {
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.scss:71]: Rename `mobileNav` to `nav-mobile`, lets be consistent with naming

File: src/components/NavBar/Dropdown.scss [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+    font-size: 13px;
+  }
+}
+
+#first-link:hover,
+#second-link:hover {
+  & > .sub-menu {
+    visibility: visible;
+  }
+}
```
Comment (VeeShort) [inline src/components/NavBar/Dropdown.scss:40]: I think we can optimize this even better:
```scss
.active-link:hover {
  & > .sub-menu {
      visibility: visible;
  }
}
```
Don't forget to remove id's from the html

File: src/components/NavBar/Dropdown.js [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+          <i className="ion-chevron-right" />
+        </span>
+        <ul className="sub-menu">
+          <li>
+            <a href="#">Service1</a>
+          </li>
+          <li>
+            <a href="#">Service2</a>
+          </li>
+          <li className="active-link" id="second-link">
```
Comment (VeeShort) [inline src/components/NavBar/Dropdown.js:41]: Remove this id. See a comment below in `Dropdown.scss`

File: src/components/NavBar/Dropdown.js [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+
+  return (
+    <ul className="sub-menu main">
+      <li>
+        <a href="#">Service1</a>
+      </li>
+      <li>
+        <a href="#">Service2</a>
+      </li>
+      <li className="active-link" id="first-link">
```
Comment (VeeShort) [inline src/components/NavBar/Dropdown.js:26]: Remove this id. See a comment below in `Dropdown.scss`

**Review by VeeShort — CHANGES_REQUESTED** [review]
New chunk of comments

File: src/components/NavBar/Dropdown.js (hunk 10 above)
Comment (VeeShort) [reply src/components/NavBar/Dropdown.js:16]: Further optimization possible

File: src/components/NavBar/NavBar.js [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+                SERVICE
+              </Link>
+              <span onisMenuActive={() => setDropDownVisibility(!isDropDownActive)}>
+                <i
+                  className={isDropDownActive ? 'ion-chevron-up' : 'ion-chevron-down'}
+                />
+              </span>
+              {isDropDownActive && <Dropdown />}
+            </li>
+            <li className="nav-item" onisMenuActive={closeMobileMenu}>
```
Comment (VeeShort) [inline src/components/NavBar/NavBar.js:57]: What is `onisMenuActive`?

**Review by VeeShort — CHANGES_REQUESTED** [review]

**Comment by VeeShort** [comment]
Remove all imports for `React`

File: src/components/NavBar/Dropdown.js (hunk 10 above)
Comment (EduardKurbatov) [reply src/components/NavBar/Dropdown.js:16]: I try to do like on Your example and I got an error about infinite loop

**Pull request merged**